browser/ui/webui/side_panel/customize_chrome/customize_toolbar/list_action_modifiers_unittest.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/list_action_modifiers.h"
+
+#include <utility>
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+using ListActionModifiersUnitTest = ::testing::Test;
+
+TEST_F(ListActionModifiersUnitTest, FilterUnsupportedChromiumAction) {
+  // We have four actions that are not supported.
+  //  * ActionId::kShowPaymentMethods
+  //  * ActionId::kShowTranslate
+  //  * ActionId::kShowReadAnything
+  //  * ActionId::kShowAddresses
+  // Checks if customize_chrome::FilterUnsupportedChromiumActions() filters them
+  // out.
+  using ActionId = side_panel::customize_chrome::mojom::ActionId;
+  side_panel::customize_chrome::mojom::ActionPtr action =
+      side_panel::customize_chrome::mojom::Action::New();
+  action->id = ActionId::kShowPaymentMethods;
+  std::vector<side_panel::customize_chrome::mojom::ActionPtr> actions;
+  actions.push_back(std::move(action));
+  action = side_panel::customize_chrome::mojom::Action::New();
+  action->id = ActionId::kShowTranslate;
+  actions.push_back(std::move(action));
+  action = side_panel::customize_chrome::mojom::Action::New();
+  action->id = ActionId::kShowReadAnything;
+  actions.push_back(std::move(action));
+  action = side_panel::customize_chrome::mojom::Action::New();
+  action->id = ActionId::kShowAddresses;
+  actions.push_back(std::move(action));
```

Comment:
I'd say this should fill the actions vector from min to max value and expect only listed actions are removed. All other should be left intact.

You can fill a vector with a single item in a for-loop to check each item explicitly, and then add a separate check with multiple items in a vector to make sure it actually removes and keeps multiple items.

This way if you add a new action to ignore, the test will actually fail.

Reply:
Good call!